manifests/client.pp

```diff
@@ -44,12 +44,35 @@
           content => template("${module_name}/sensu-client.erb"),
         }
 
-        exec { 'install-sensu-client':
-          provider => 'powershell',
-          command  => "New-Service -Name sensu-client -BinaryPathName c:\\opt\\sensu\\bin\\sensu-client.exe -DisplayName 'Sensu Client' -StartupType Automatic",
-          unless   => 'if (Get-Service sensu-client -ErrorAction SilentlyContinue) { exit 0 } else { exit 1 }',
-          before   => Service['sensu-client'],
-          require  => File['C:/opt/sensu/bin/sensu-client.xml'],
+        if $::sensu::windows_service_user {
+          dsc_userrightsassignment { $::sensu::windows_service_user['user']:
+            dsc_ensure   => present,
+            dsc_policy   => 'Log_on_as_a_service',
+            dsc_identity => $::sensu::windows_service_user['user'],
+            before       => Dsc_service['sensu-client'],
+          }
+
+          acl { 'C:/opt/sensu':
+            purge       => false,
+            permissions => [
+              {
+                'identity' => $::sensu::windows_service_user['user'],
+                'rights'   => ['full'],
+              },
+            ],
+            before      => Dsc_service['sensu-client'],
+          }
+        }
+
+        dsc_service { 'sensu-client':
+          dsc_ensure      => present,
+          dsc_name        => 'sensu-client',
+          dsc_credential  => $::sensu::windows_service_user,
+          dsc_displayname => 'Sensu Client',
+          dsc_path        => 'c:\\opt\\sensu\\bin\\sensu-client.exe',
+          require         => File['C:/opt/sensu/bin/sensu-client.xml'],
+          # See MODULES-4570
```

Comment:
Please update comment with the link to the full URL - https://tickets.puppetlabs.com/browse/MODULES-4570

Reply:
besides the link, explaining above the dsc_service that you need it and the service resource because of the bug would be helpful as it is not intuitive that you would have two service related resources.

Reply:
Add explanation that describes how `dsc_service` just installs the service but `Service[sensu-client]` handles state and refreshes.